File: U2/Demos/U2-Toolkit/.NET Samples/samples/VB.NET/UniVerse/EntityFramework_CodeFirst/Customer.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Entity

Namespace EntityFramework_CodeFirst

    <Table("CUSTOMER", Schema:="HS_SALES")> _
    Public Class Customer
        Sub New()
            Orders = New HashSet(Of Order)()
        End Sub

        <Key()> _
        Public Property CUSTID() As Integer

        Public Property FNAME() As String

        Public Overridable Property Orders() As ICollection(Of Order)
    End Class

    <Table("CUSTOMER_ORDERS", Schema:="HS_SALES")> _
    Public Class Order
        Sub New()
        End Sub


        <Key()> _
        <Column("@ASSOC_ROW")> _
        Public Property ASSOC_ROW() As Decimal

        Public Property CUSTID() As Integer
        Public Property PRODID() As String
        Public Property SER_NUM() As String

        Public Overridable Property Customer() As Customer

    End Class
    Public Class CustomerContext
        Inherits DbContext

        Public Sub New()
        End Sub

        Public Property Customers() As DbSet(Of Customer)
        Public Property Orders() As DbSet(Of Order)
        Protected Overrides Sub OnModelCreating(modelBuilder As DbModelBuilder)

            modelBuilder.Entity(Of Customer)().[Property](Function(s) s.FNAME).IsRequired()
        End Sub


    End Class

End Namespace

File: U2/Demos/U2-Toolkit/.NET Samples/samples/C#/UniVerse/EntityFramework_CodeFirst/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;

namespace EntityFramework_CodeFirst
{

    //[Table("CUSTOMER", Schema = "HS_SALES")] //UNDK-24
    [Table("CUSTOMER")]
    public class Customer
    {
        Customer()
        {
            Orders = new HashSet<Order>();
        }
        [Key]
        public int CUSTID { get; set; }

        public string FNAME { get; set; }

        public virtual ICollection<Order> Orders { get; set; }
    }

    //[Table("CUSTOMER_ORDERS", Schema = "HS_SALES")]
    [Table("CUSTOMER_ORDERS")]
    public class Order
    {
        Order()
        {
        }

        [Key]
        [Column("@ASSOC_ROW")]
        public decimal @ASSOC_ROW { get; set; }
        
        public int CUSTID { get; set; }
        public string PRODID { get; set; }
        public string SER_NUM { get; set; }

        public virtual Customer Customer { get; set; }

    }
    public class CustomerContext : DbContext
    {
        public CustomerContext()
        {
            
        }
        public DbSet<Customer> Customers { get; set; }
        public DbSet<Order> Orders { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Customer>()
                .Property(s => s.FNAME)
                .IsRequired();
            modelBuilder.Conventions.Remove<IncludeMetadataConvention>(); 
        }

    }



}
